### DISCUSSION TITLE
compiling error for lammps using "make"

### DISCUSSION THREAD

**Discussion by jushinpon** [deepmodeling/deepmd-kit#3259] [Q&A]
Dear DeepMD developers,
I use a Rocky Linux 9 server to install lammps with DeePMD following the instruction of the DeepMD-kit  document. I first used conda to establish deepmd-cpu environment (adopt the easy install process). Within this environment (deepmd-cpu), I can successfully Install the DeePMD-kit’s C++ interface and get all required files in USER-DEEPMD. The command for cmake is:
cmake -DENABLE_TENSORFLOW=TRUE -DUSE_TF_PYTHON_LIBS=TRUE -DLAMMPS_SOURCE_ROOT=/home/packages/lammps4deepmd/lammps-stable_2Aug2023_update2 -DCMAKE_INSTALL_PREFIX=/opt/deepmd_lammpslib/ ..

The output of cmake indicate the lammps plugin mode has been enabled:

-- enable LAMMPS plugin mode
-- LAMMPS_HEADER_DIR is /home/packages/lammps4deepmd/lammps-stable_2Aug2023_update2/src
-- Found MPI_C: /opt/mpich-4.0.3/lib/libmpi.so (found version "4.0") 
-- Found MPI_CXX: /opt/mpich-4.0.3/lib/libmpicxx.so (found version "4.0") 
-- Found MPI: TRUE (found version "4.0")  
-- LAMMPS version is 20230802

However,  when I use make to install lammps, the following error shows:

In file included from ../packages_pair.h:1572,
                 from ../lammps.cpp:1111:
../USER-DEEPMD/pair_deepmd.h:3:2: error: #error Please define LAMMPS_VERSION_NUMBER to yyyymmdd
    3 | #error Please define LAMMPS_VERSION_NUMBER to yyyymmdd

Would you please instruct me what I should do for fixing this error. Appreciate your kind help in advance. By the way, this lammps compilation process is ok for Rocky linux 8 server.

Shin-Pon Ju 
Professor
Department of Mechanical and
Electro-Mechanical Engineering
National Sun Yat-Sen University
Kaohsiung, Taiwan, Republic of China
Office: 886-7-5252000 Ext. 4231
fax: 886-7-5252132
Cellular Phone: 886-929579443
E-mail: [jushin-pon@mail.nsysu.edu.tw](mailto:jushin-pon@mail.nsysu.edu.tw)
http://www2.nsysu.edu.tw/mel/

**Comment by njzjz** [comment]
Dear Prof. Ju,
It seems you mixed the plugin mode and the built mode. The `USER-DEEPMD`
directory is for the built-in mode. The CMake argument
`-DLAMMPS_SOURCE_ROOT` is for the plugin mode. The plugin mode doesn't
support Make.
The built-in mode with Make is okay, but you need to execute `make
yes-user-deepmd` before building LAMMPS.
I suggest writing all scripts that you executed, which will make it much
easier to figure out what is wrong.

**Reply by jushinpon** [reply]
Thank you very much for your fast reply and instructions. I will follow your suggestion to make lammps executable for our Rocky Linux 9 cluster and let you know the result.
